addons/website/models/ir_http.py

```diff
@@ -104,9 +103,12 @@ def _generate_routing_rules(cls, modules, converters):
 
                     if url != url_to:
                         logger.debug('Redirect from %s to %s for website %s' % (url, url_to, website_id))
-                        _slug_matching = partial(cls._slug_matching, endpoint=endpoint)
-                        endpoint.routing['redirect_to'] = _slug_matching
-                        yield url, endpoint  # yield original redirected to new url
+                        # duplicate the endpoint to only register the redirect_to for this specific url
+                        redirect_endpoint = functools.partial(endpoint)
+                        functools.update_wrapper(redirect_endpoint, endpoint)
```

Comment:
This is just about making it easier to debug should we need to, regarding the callstack etc, right?

Reply:
Yes, this line copies the few `__help__`, `__name__`, ... of the original method onto the partialized one so inside the traceback it looks like we called the original method. Users don't need to know that their actual controller method was partialized... twice... (this trick of using a partial is also present in http._generate_routing_map).

But it also copies all attributes that were manually set on the function:

```py
def foo():
    pass
foo.x = 5

import functools
bar = functools.partial(foo)
getattr(bar, 'x', None)  # None

functools.update_wrapper(bar, foo)
bar.x  # 5
```

and this is important because we need to copy `endpoint.original_method` that is used in a few website methods

Reply:
So it is *both* nice to have *and* required :p

Reply:
Oh, ok. Thank you for the explanation :+1: